src/presentation/http/router/note.test.ts

```diff
@@ -521,6 +542,139 @@ describe('Note API', () => {
     });
   });
 
+  describe('PATCH /note/:notePublicId/parent', () => {
+    test('Returns 200 when parent was successfully updated', async () => {
+      const childNotePublicId = '9OYDD9d4_Y';
+
+      const parentNotePublicId = 'Tocn1f7rQS';
+
+      const newParentNotePublicId = 'lMkesQ1gpg';
+
+      await global.db.truncateTables();
+
+      const creator = await global.db.insertUser({
+        email: 'a@a.com',
+        name: 'test user',
+      });
+
+      const childNote = await global.db.insertNote({
+        creatorId: creator.id,
+        publicId: childNotePublicId,
+      });
+
+      const parentNote = await global.db.insertNote({
+        creatorId: creator.id,
+        publicId: parentNotePublicId,
+      });
+
+      await global.db.insertNote({
+        creatorId: creator.id,
+        publicId: newParentNotePublicId,
+      });
+
+      await global.db.insertNoteRelation({
+        noteId: childNote.id,
+        parentId: parentNote.id,
+      });
+
+      const accessToken = global.auth(creator.id);
+
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        body: {
+          parentId: newParentNotePublicId,
+        },
+        url: `/note/${childNotePublicId}/parent`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      expect(response?.json()).toStrictEqual({ isUpdated: true });
+    });
+
+    test('Returns 500 when parentId is the same as childId', async () => {
```

Comment:
Why 500? We should never have an answer of 5xx
You need to find the error within the router